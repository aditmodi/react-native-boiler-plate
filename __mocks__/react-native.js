import React, { Component } from 'react';

export const ReactNative = React;

ReactNative.StyleSheet = {
  create: function create(styles){
    return styles
  }
}

class View extends Component{
  render(){ return false; }
}

class ListView extends Component{
  static DataSource(){}
}

class AppRegistry{
  static registerComponent(){}
}

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TouchableOpacity = View;
ReactNative.TouchableHighlight = View;
ReactNative.TouchableWithoutFeedback = View;
ReactNative.ToolbarAndroid = View;
ReactNative.Image = View;
ReactNative.AppRegistry = AppRegistry;
