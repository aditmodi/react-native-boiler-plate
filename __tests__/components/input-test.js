import 'react-native';
import React from 'react';
import InputField from '../../app/components/input';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <InputField/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
