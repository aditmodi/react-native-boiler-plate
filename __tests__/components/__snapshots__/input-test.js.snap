// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "marginLeft": 2,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "left": 0,
          "opacity": 1,
          "paddingBottom": undefined,
          "paddingTop": undefined,
          "position": "absolute",
          "right": 0,
          "top": 18,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#575757",
            "fontSize": 17,
            "paddingRight": 5,
            "top": 8,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 17,
            }
          }
          virtual={undefined}
        />
      </Text>
    </View>
    <TextInput
      editable={true}
      keyboardType={null}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#575757"
      secureTextEntry={null}
      style={
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 60,
          "lineHeight": 24,
          "paddingLeft": 5,
          "paddingRight": 5,
          "top": 8,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
      virtual={undefined}
    />
  </View>
</View>
`;
